src/MailboxesParser/MailboxesParser.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Lukasz
+ * Date: 2017-11-10
+ * Time: 19:49
+ */
+
+namespace Ddeboer\Imap\MailboxesParser;
+
+
+use Ddeboer\Imap\Exception\MailboxesParserException;
+use Ddeboer\Imap\MailboxInterface;
+
+class MailboxesParser
+{
+    /** @var MailboxInterface[] */
+    protected $mailboxes;
+    /** @var ParsedMailbox[] */
+    protected $folders;
+    protected $treeStructure;
+
+    const INBOX = 'inbox';
+    const SENT = 'sent';
+    const DRAFT = 'drafts';
+    const SPAM = 'spam';
+    const TRASH = 'trash';
+    const TEMPLATES = 'templates';
+    const ARCHIVES = 'archives';
+
+    protected $specialFoldersIds = [
+        self::INBOX     => ['inbox',],
+        self::SENT      => ['sent', 'sent messages', 'INBOX.Sent', '[Gmail]/Sent Mail',],
+        self::DRAFT     => ['drafts', 'INBOX.Drafts', '[Gmail]/Drafts',],
+        self::SPAM      => ['spam', 'INBOX.spam', '[Gmail]/spam'],
+        self::TRASH     => ['trash', 'bin', 'INBOX.trash', '[Gmail]/trash'],
+        self::TEMPLATES => ['templates'],
+        self::ARCHIVES  => ['archives',],
+    ];
+
+    protected $specialFoldersNames = [
+        self::DRAFT     => 'Drafts',
+        self::INBOX     => 'Inbox',
+        self::SENT      => 'Sent',
+        self::SPAM      => 'Spam',
+        self::TRASH     => 'Trash',
+        self::TEMPLATES => 'Templates',
+        self::ARCHIVES  => 'Archives',
+    ];
+
+    protected $specialFoldersOrder = [
+        self::INBOX     => 1,
+        self::SENT      => 2,
+        self::DRAFT     => 3,
+        self::TEMPLATES => 4,
+        self::ARCHIVES  => 10000,
+        self::SPAM      => 20000,
+        self::TRASH     => 30000,
+    ];
+
+    /**
+     * MailboxesTree constructor.
+     *
+     * @param MailboxInterface[] $mailboxes
+     */
+    public function __construct($mailboxes)
```

Comment:
Please add type-hints

Reply:
I think in php 7.0 I cannot use `public function __construct(MailboxInterface[] $mailboxes)`

Reply:
No, but you can typehint the `array` at least.

Reply:
I did it

Reply:
I still see `__construct($mailboxes)`.

Can you change this to `__construct(array $mailboxes)`?